### PR TITLE
Update ruby percy client to use git data if possible, then fallback to environment variables

### PR DESCRIPTION
**Pull request by cadeParade**
Refactor commit method to:
-  Check if we can get data from git.
    - If so, parse git data and return that.
    - If not, try to get some data from environment variables and return that.

Update tests to test specifically for data in git/environment specifically.

### CODE REVIEW

File: lib/percy/client/environment.rb [hunk 1]
```diff
@@ -28,33 +28,48 @@ def self.current_ci
...
       # @return [Hash] All commit data from the current commit. Might be empty if commit data could
       # not be found.
+      # Try getting data from git itself.
+      # If git is not present, fallback on data from environment variables.
       def self.commit
-        output = _raw_commit_output(_commit_sha) if _commit_sha
-        output ||= _raw_commit_output('HEAD')
-        return {branch: branch} unless output && output != ''
-        output = output.force_encoding('UTF-8') if output && output.encoding.to_s == 'US-ASCII'
+        git_data_from_git = _raw_commit_output(_commit_sha || 'HEAD')
```
Comment (timhaines) [inline lib/percy/client/environment.rb]: This is not logically equivalent.  Lets be sure to retry the command with `HEAD` if it fails with the `_commit_sha`

File: lib/percy/client/environment.rb [hunk 2]
```diff
@@ -28,33 +28,48 @@ def self.current_ci
...
 
-        # If not running in a git repo, allow nils for certain commit attributes.
-        parse = ->(regex) { (output && output.match(regex) || [])[1] }
-        {
-          # The only required attribute:
-          branch: branch,
-          # An optional but important attribute:
-          sha: commit_sha,
+        if git_data_from_git
+          parse = ->(regex) { (git_data_from_git && git_data_from_git.match(regex) || [])[1] }
```
Comment (timhaines) [inline lib/percy/client/environment.rb]: Could move parse definition up, and use it for setting commit_sha above

Comment (cadeParade) [reply lib/percy/client/environment.rb]: done!

File: lib/percy/client/environment.rb (hunk 1 above)
Comment (cadeParade) [reply lib/percy/client/environment.rb]: Done!

**Comment by cadeParade** [comment]
The Travis tests are failing because of a linter error on a method I didn't change...one of the arguments to `respond_to_missing` is too short (less than 3 characters). I can rename the parameter to something else, or ignore the line or?

**Comment by timhaines** [comment]
@cadeParade The linter problem came up because rubocop just got upgraded to include a new rule.  We don't lock the version of rubocop here, or in percy-style apparently.  I'll update percy-style to block that rule.

**Pull request merged**